android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.java

```diff
@@ -127,27 +125,10 @@ public void onDropViewInstance(ReactExoplayerView view) {
 
     @ReactProp(name = PROP_DRM)
     public void setDRM(final ReactExoplayerView videoView, @Nullable ReadableMap drm) {
-        if (drm != null && drm.hasKey(PROP_DRM_TYPE)) {
-            String drmType = ReactBridgeUtils.safeGetString(drm, PROP_DRM_TYPE);
-            String drmLicenseServer = ReactBridgeUtils.safeGetString(drm, PROP_DRM_LICENSE_SERVER);
-            ReadableArray drmHeadersArray = ReactBridgeUtils.safeGetArray(drm, PROP_DRM_HEADERS);
-            if (drmType != null && drmLicenseServer != null && Util.getDrmUuid(drmType) != null) {
-                UUID drmUUID = Util.getDrmUuid(drmType);
-                videoView.setDrmType(drmUUID);
-                videoView.setDrmLicenseUrl(drmLicenseServer);
-                if (drmHeadersArray != null) {
-                    ArrayList<String> drmKeyRequestPropertiesList = new ArrayList<>();
-                    for (int i = 0; i < drmHeadersArray.size(); i++) {
-                        ReadableMap current = drmHeadersArray.getMap(i);
-                        String key = current.hasKey("key") ? current.getString("key") : null;
-                        String value = current.hasKey("value") ? current.getString("value") : null;
-                        drmKeyRequestPropertiesList.add(key);
-                        drmKeyRequestPropertiesList.add(value);
-                    }
-                    videoView.setDrmLicenseHeader(drmKeyRequestPropertiesList.toArray(new String[0]));
-                }
-                videoView.setUseTextureView(false);
-            }
+        DRMProps drmProps = DRMProps.parse(drm);
+        if (drmProps != null) {
+            videoView.setDrm(drmProps);
```

Comment:
nit; If the drm prop is inserted as null, the old drm prop will still be there. wouldn't it be a problem? (I think it's a problem from the old behavior, not from this change)

Reply:
No It was not the goal of this PR, So I remove the test.
It doesn't make sense to keep previous DRM config when user remove it !
Thank you for the catch !